src/main/frontend/db/query_dsl.cljs

```diff
@@ -315,6 +322,36 @@
                                                  (string/join " ")
                                                  (util/format "(between %s)"))))))
 
+(defn- add-bindings!
+  [q]
+  (let [b? (atom false)
+        p? (atom false)
+        not? (atom false)]
```

Comment:
This can be made functional by

```clojure
((juxt (partial some #(= '?not %))
       (partial some #(= '?b   %))
       (partial some #(= '?p   %)))
 (flatten q))
```

Reply:
:+1: juxt && flatten is much better than introducing mutable refs here. But I think a normal `set/intersection` with `flatten` should be easier to read.

Reply:
Agreed, and `juxt` here actually iterates through the sequence 3 times.